### PR TITLE
biguintcore: eliminated unnecessary divides

### PR DESCRIPTION
**Pull request by WalterBright**
And eliminate unused parameter.

### CODE REVIEW

**Comment by burner** [comment]
why did you remove "radix", it was used

**Comment by WalterBright** [comment]
If it was used, Phobos would fail to compile :-)

**Comment by burner** [comment]
bugzilla issue id?

**Comment by JackStouffer** [comment]
Do you really need one for optimizations?

**Comment by WalterBright** [comment]
> bugzilla issue id?

?? This PR is not a bug fix.

**Comment by burner** [comment]
I think we misunderstand each other.

Did it fail before, you created this PR, or would it fail with this PR if you didn't remove radix?

File: std/internal/math/biguintcore.d [hunk 1]
```diff
@@ -2161,14 +2161,21 @@ again:
...
 
 // TODO: Replace with a library call
-void itoaZeroPadded(char[] output, uint value, int radix = 10)
-    pure nothrow @safe
+void itoaZeroPadded(char[] output, uint value)
+    pure nothrow @safe @nogc
 {
-    ptrdiff_t x = output.length - 1;
-    for( ; x >= 0; --x)
+    for (auto i = output.length; i--;)
```
Comment (JackStouffer) [inline std/internal/math/biguintcore.d]: Why not `foreach_reverse(i, item; output)`?

**Comment by WalterBright** [comment]
If I removed a parameter, and someone called it using that parameter, it would fail to compile Phobos. Since Phobos does compile and pass all its unit tests, then that parameter was not used.

I.e. you said it was used => show me!

File: std/internal/math/biguintcore.d (hunk 1 above)
Comment (WalterBright) [inline std/internal/math/biguintcore.d]: because I don't like foreach_reverse.

**Comment by burner** [comment]
I'm still not sure if we are talking about the same thing, but I just now saw the "private: " above the function (aka no public api breakage)

Therefore, LGTM

**Comment by burner** [comment]
Auto-merge toggled on

**Comment by WalterBright** [comment]
Thanks!

**Comment by burner** [comment]
welcome

**Pull request merged**